tests/unit/development/test_r_component_gen.py

```diff
@@ -0,0 +1,85 @@
+import os
+import shutil
+import re
+from textwrap import dedent
+
+import pytest
+
+from dash.development._r_components_generation import (
+    make_namespace_exports
+)
+
+
+@pytest.fixture
+def make_r_dir():
```

Comment:
you can use tmpdir, it's built-in fixture from pytest

Reply:
Except that as the code stands we need the dir to be `./R`, and AFAICT `tmpdir` creates its own and you can't control it. I suppose I could make the R source path a param to `make_namespace_exports`... not sure that's worth it though.

Reply:
that's not true, we have no control of the temp parent folder name, but  `tmpdir.mkdir("R")` :)

Reply:
anyway, this is minor, and you have my 💃

Reply:
Mmm, I suppose, then we `os.chdir` into the `tmpdir`, and back out when we're done. If you feel like cleaning this up after I merge go for it :)